Sources/ArcGISToolkit/Components/FeatureTemplatePicker/FeatureTemplatePicker.swift

```diff
@@ -0,0 +1,365 @@
+        }
+        /// The complete unfiltered list of feature template sections.
+        private(set) var unfilteredFeatureTemplateSections = [FeatureTemplateSectionInfo]() {
+            didSet { templatesOrSearchTextDidChange() }
+        }
+        
+        /// The sections.
+        @Published private(set) var featureTemplateSections = [FeatureTemplateSectionInfo]()
+        /// A Boolean value indicating whether templates are being generated.
+        @Published private(set) var isGeneratingFeatureTemplates = false
+        /// A Boolean value indicating if content is unavailable.
+        @Published private(set) var showContentUnavailable = false
+        /// A Boolean value indicating if no templates were found with the given search text.
+        @Published private(set) var showNoTemplatesFound = false
+        
+        /// Creates a model for a given geo model.
+        init(geoModel: GeoModel, includeNonCreatableFeatureTemplates: Bool) {
+            self.geoModel = geoModel
+            self.includeNonCreatableFeatureTemplates = includeNonCreatableFeatureTemplates
+        }
+        
+        /// Generates the feature templates.
+        func generateFeatureTemplates() async {
+            isGeneratingFeatureTemplates = true
+            defer { isGeneratingFeatureTemplates = false }
+            
+            unfilteredFeatureTemplateSections = await makeFeatureTemplateSections()
+        }
+        
+        /// The feature template section information for the geo model.
+        private func makeFeatureTemplateSections() async -> [FeatureTemplateSectionInfo] {
+            try? await geoModel.load()
+            var sections = [FeatureTemplateSectionInfo]()
+            let layersAndTables = geoModel.arcGISFeatureLayersAndTables
+            await layersAndTables.map(\.table).load()
+            for (layer, table) in layersAndTables {
+                guard includeNonCreatableFeatureTemplates || table.canAddFeature,
+                      // If the table failed to load then makeFeature will precondition fail.
+                      table.loadStatus == .loaded
+                else { continue }
+                var infos = [FeatureTemplateInfo]()
+                for template in table.allTemplates {
+                    let feature = table.makeFeature(template: template)
+                    let symbol = layer.renderer?.symbol(for: feature)
+                    let scale: CGFloat
+#if os(visionOS)
+                    scale = 1
+#else
+                    scale = UIScreen.main.scale
+#endif
+                    let image = try? await symbol?.makeSwatch(scale: scale)
+                    infos.append(
+                        FeatureTemplateInfo(
+                            layer: layer,
+                            table: table,
+                            template: template,
+                            image: image
+                        )
+                    )
+                }
```

Comment:
Should the templates be sorted alphabetically by name?

```suggestion
                    )
                    infos.sort { $0.template.name < $1.template.name }
                }
```

Reply:
Perhaps the sections too?

Reply:
I'm not sure. I could see alphabetizing them. Currently it is in the order of layers and then the templates that the layers report. So I'm not sure what is desirable. @mhdostal do you recall what the obj-c based version did?

Reply:
> I'm not sure. I could see alphabetizing them. Currently it is in the order of layers and then the templates that the layers report. So I'm not sure what is desirable. @mhdostal do you recall what the obj-c based version did?

There was no sorting in the obj-c template picker.